More moves toward using new net_pkt API

tbursztyka opened the pull request:
basically going ahead by removing as much as legacy API as possible

Commit:
net/sockets: Adapt net_pkt_get_src_addr to new net_pkt API

This function is only used in sockets, thus making it a private function
of socket library and renaming it relevantly.

Note that sockets should be reviewed at some point to avoid using such
function: zsock_received_cb() already get the ip header and the protocol
header, so it could grab the src addr/port from there. It would be way
more optimized to do so, since net_pkt_get_src_addr is costly as it
parses all over again the ip/protocol headers.

utils unit test is updated and the test of the former
net_pkt_get_src_addr/net_pkt_get_dst_addr are removed.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/pkt: Removing transport protocol attribute

It is now unused anywhere: former net_pkt_get_src/dst_addr where the
only one using it and that has been changed since.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/tcp: Move net_tcp_get_hdr() away from net core

Only the unit test needs it now, so let's move it there.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/pkt: Fix legacy and new allocator attribute initialization

Legacy net_pkt allocator was not setting default value for IPv6
next header. And new net_pkt allocator was not setting default value
for TX traffic class and virtual LAN tag.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/context: Let's use the new net_ipv6_finalize

And removing the legacy and unused one.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/utils: Backup the cursor at relevant place in chksum calculation

If the family of the packet is not recognized, then do not backup the
cursor.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/context: Let's use the new net_ipv4_finalize

And removing the legacy and unused one.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/ipv4: Always return result of the last finalizer

Or we will not know if overall finalization worked.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/ipv6: Always return result of the last finalizer

Or we will not know if overall finalization worked.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/ipv6: Rename net_ipv6_finalize since legacy version disappeared

s/net_ipv6_finalize_new/net_ipv6_finalize

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/ipv4: Rename net_ipv4_finalize since legacy version disappeared

s/net_ipv4_finalize_new/net_ipv4_finalize

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/ipv6: No need to finalize icmpv6 twice in MLD generation

net_ipv6_finalize() can take care of this, if net_pkt's ipv6_next_hdr
attribute is set to the right one, as well as ipv6_ext_len one.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/tcp: Removing net_tcp_set_chksum function

Now that it's unused, it's useless.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/tcp: Move net_tcp_set_hdr() away from net core

Only the unit test needs it now, so let's move it there.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/tcp: Remove useless tcp function using legacy net_pkt API

net_pkt_tcp_data was only used in tcp unit test and could be replaced by
local net_tcp_get_hdr.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/icmp: Remove useless icmp functions using legacy net_pkt API

net_pkt_icmp_data was only used in ipv6 unit test and could be easily
replaced.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/icmpv6: Remove useless net_icmpv6_set_chksum

net_ipv6_finalize do not use it anymore.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/udp: Remove useless net_udp_set_chksum

It was only used in ipv6_fragment unit test and could be easily
replaced.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

tbursztyka commented on subsys/net/ip/utils.c:
```diff
@@ -766,3 +766,104 @@ int net_bytes_from_str(u8_t *buf, int buf_len, const char *src)
 	return 0;
 }
+
+int net_pkt_get_src_addr(struct net_pkt *pkt,
```
It is a utility function. A very temporary one actually, because we should get rid of it asap as stated in the commit message. Only socket uses it because it's not taking advantage of net_context's recv_cb ip_hdr, proto_hdr parameters. I really don't want to see anybody else using that function, actually I might move it into socket directly

jukkar replied:
Sure, having the prototype in net_private.h is ok, it is just that moving the implementation to utils.c seems a bit pointless here, we can keep it in net_pkt.c as the function name indicates.

tbursztyka replied:
I was more radical: I move the function in socket.c and that's all. It's unused anywhere else, and removes code to maintain in utils unit test. And hopefully in some near future, socket will not need that function anymore as well.

tbursztyka commented on subsys/net/ip/net_pkt.c:
```diff
@@ -359,6 +359,10 @@ struct net_pkt *net_pkt_get_reserve(struct k_mem_slab *slab,
 	pkt->atomic_ref = ATOMIC_INIT(1);
```
indeed,

Commit:
net/udp: Switch net_udp_get/set_hdr to new net_pkt API

Let's use the new API for these 2 functions. Note that in some places,
using these functions is under-optimized (like in llmnr-responder in
dns: it gets the ip/udp headers already from the recv callback. That
will need to be taken care of later).

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/udp: Remove net_pkt_udp_data() function

It is now useless and can be replaced by net_udp_get_hdr() directly, in
the 2 unit tests it was used.

Removing as well the dbg function too_short_msg()

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/udp: Remove net_udp_header_fits legacy function.

It not used anymore. If one wants to do the same check, it will require
to place the net_pkt cursor at the relevant position and use
net_pkt_is_contiguous(pkt, sizeof(struct net_udp_hdr))

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/utils: Remove net_header_fits() function

It is not used anymore. Instead, use net_pkt_is_contiguous()

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/private: Rename net_hexdump_frags to net_pkt_hexdump

And remove the parameter "full" as there is no "ll reserve" distinction
anymore. The parameter was unused since the ll reserve concept removal.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/private: Rename net_print_frags() function

Though unused anywhere, this function seems to have some usage while
debugging.

Let's rename it to a more semantically relevant name.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/udp: Let's set the appdata related attributes once header parsed

No need to do it later as all necessary informations are there already.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/tcp: Let's set the appdata related attributes by ourselves

Let's not call net_pkt_set_appdata_values() which will be more costly
since it will need to parse all over again the packet to grab the tcp
header. Instead, let's use the tcp header pointer we have already and
set the appdata attributes directly.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/context: Do not set the appadata attributes again

These are already set either by udp or tcp at relevant places.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/pkt: Remove useless net_pkt_set_appdata_values() function

UDP/TCP are setting the appdata attributes by themselves now.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/tcp: Remove useless tcp_hdr_len() function

It is now useless as the only function using it
(net_pkt_set_appdata_values) got removed.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/socket: Use the new net_pkt API to read UDP payload

There is no need for appdata pointer: net_pkt's cursor is already at the
right position, i.e. the beginning of the payload right after all IP/UDP
headers.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/pkt: Add a function to know the amount of data to read left

It will return the amount of data to be read from current cursor
position. This will prove to be useful to remove appdatalen attribute.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/socket: Use the new net_pkt API to read TCP payload

There is no need for appdata pointer: net_pkt's cursor is already at the
right position, i.e. the beginning of the payload right after all IP/TCP
headers.

Also, when reading the actual data, let's use net_pkt_read_new()
relevantly instead of going through the buffer by ourselves.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

GeorgeCGV commented:
Is it intended that new `send` and `sendto` completely ignore `CONFIG_NET_OFFLOAD`?

tbursztyka commented:
@GeorgeCGV  for now yes. It's on my task list.

GeorgeCGV commented:
> @GeorgeCGV for now yes. It's on my task list.

Is it planned at least for the initial release of `1.14`?

tbursztyka commented:
@GeorgeCGV  yes

tbursztyka commented:
@jukkar , @pfalcon and @rveerama1 : can you (re)review ? Would be nice to push this.

rveerama1 commented on subsys/net/lib/sockets/sockets.c:
```diff
@@ -517,24 +614,20 @@ static inline ssize_t zsock_recv_dgram(struct net_context *ctx,
-	 */
-	(void)net_frag_linearize(buf, recv_len, pkt, header_len, recv_len);
+	if (net_pkt_read_new(pkt, buf, recv_len)) {
+		errno = -ENOBUFS;
```
Remove that "-" from -ENOBUFS. errno is not -ve.

tbursztyka replied:
good catch

pfalcon commented on subsys/net/lib/sockets/sockets.c:
```diff
@@ -459,6 +459,107 @@ Z_SYSCALL_HANDLER(zsock_sendto, sock, buf, len, flags, dest_addr, addrlen)
+	}
+
+	net_pkt_cursor_backup(pkt, &backup);
+	net_pkt_cursor_init(pkt);
```
So, this is first time I see this "packet cursor" thing (I like the name, hope the implementation is unbloated). So, I know nothing about it. But I find strange that we first backup packet cursor, then initialize it. Either they're swapped, or there's naming problem, e.g. "net_pkt_cursor_init" is actually "net_pkt_cursor_reset" (i.e. resets cursor to the beginning of a packet).

tbursztyka replied:
I am not going to do any change name right now. semantically it's not that much different init vs reset.

both mean the cursor is set at the beginning of net_pkt buffer (if there is one).

pfalcon commented on include/net/net_pkt.h:
```diff
@@ -2022,32 +2022,6 @@ void net_pkt_get_info(struct k_mem_slab **rx,
 		      struct net_buf_pool **rx_data,
```
To piggyback here on commit message/overall commit matters.

> This function is only used in sockets, thus making it a private function
of socket library and renaming it relevantly.

So, it was done like it was done to have a proper layering: sockets are a layer above native API, and uses the native API to access packets. As we have an API to access packet info, it only made sense to add net_pkt_get_src_addr() to it, even if sockets are the only user (so far).

It's probably OK now to move it to sockets, but if that move leads to loss of tests for that API, I'd rather be -1 on this commit. Can you at least also adapt, not remove the tests?

pfalcon reviewed (commented):
I find "net/sockets: Adapt net_pkt_get_src_addr to new net_pkt API" commit problematic due to the loss of elaborated tests.

pfalcon commented on include/net/net_pkt.h:
```diff
@@ -165,6 +155,10 @@ struct net_pkt {
 					     */
 	};
 
+#if defined(CONFIG_NET_TCP)
```
Dunno why this was moved around.

tbursztyka commented on tests/net/utils/src/main.c:
```diff
@@ -108,69 +108,6 @@ static const unsigned char pkt3[199] = {
-/* IPv6 TCP (224 bytes) */
-static const unsigned char v6_tcp_pkt1[224] = {
-/* IPv6 header starts here */
-0x60, 0x00, /* .J....`. */
```
net_pkt_get_src/dst disappearing there is no need for any test

tbursztyka replied:
That test was relevant because of the existence of net_pkt_get_src/dst_addr functions. But these are removed. (and should never come back actually, or that will mean we do something wrong in the core).

pfalcon commented on subsys/net/ip/net_context.c:
```diff
@@ -1016,7 +1016,11 @@ static int create_udp_packet(struct net_context *context,
 		pkt = tmp;
 
-		r = net_ipv6_finalize(pkt, net_context_get_ip_proto(context));
+		net_pkt_compact(pkt);
```
Oh no, net_pkt_compact() is still there? ;-)

tbursztyka replied:
yes still part of the legacy api (create_udp_packet() here is also legacy btw)

pfalcon commented on subsys/net/ip/ipv4.c:
```diff
@@ -132,15 +132,13 @@ int net_ipv4_finalize_new(struct net_pkt *pkt, u8_t next_header_proto)
 			return net_udp_finalize(pkt);
```
Here, the commit title "net/ipv4: Always return from the last finalizer" is also not very clear (as if the alternative was to infinite loop in it, or what?). Better: "net/ipv4: Always return **result of** the last finalizer"

tbursztyka replied:
would sound better yes

tbursztyka commented:
> It's probably OK now to move it to sockets, but if that move leads to loss of tests for that API, I'd rather be -1 on this commit. Can you at least also adapt, not remove the tests?

Re-read my commit message: socket will need to be reworked a bit to actually get rid completely of such function. It's a bad thing to re-parse net_pkt all over again: it got already parsed by the core, and the info provided to net_context's recv_cb so we will need to modify socket to use the info from there.

I don't know how much you reviewed in the previous PR now merged in master introducing the new net_pkt api: but it removed a huge amount of re-parsing the net_pkt. It was everywhere, and that was super slow.

So no, that temporary function (because it is temporary) will stay in socket.

On a side note, there is no need to be scared of test removal: because if the pkt ended up to net_context's recv_cb it means the core parsed the pkt successfully, and set the right info (ip_hdr_len, ext_hdr_len etc...) so this function will get the info properly.

pfalcon commented on include/net/net_pkt.h:
```diff
@@ -2399,6 +2399,15 @@ static inline int net_pkt_write_le32_new(struct net_pkt *pkt, u32_t data)
+ *
+ * @return Amount of data to be read from current pkt cursor
+ */
+size_t net_pkt_data_to_read(struct net_pkt *pkt);
```
Can this be named more clearly, e.g. net_pkt_remaining_data_len() or net_pkt_rest_data_len() or net_pkt_tail_data_len()

tbursztyka replied:
hum, neither nor :)
I am not also super happy of the name I came from. The point here is to get the length of data in front of the current cursor position.

I am open to ideas

pfalcon replied:
> The point here is to get the length of data in front of the current cursor position.

That formulation is itself confusing. When you stand *in front of a door*, you stand *before a door*. Here, we talk about data *after* cursor.

I don't see how you don't like any of the 3 options proposed. Call it "net_pkt_data_len_after_cursor()" then, can't beat that in clarity, not so in terseness.

tbursztyka replied:
still, not great. I am sure we can find something better

pfalcon commented on include/net/net_pkt.h:
```diff
@@ -2399,6 +2399,15 @@ static inline int net_pkt_write_le32_new(struct net_pkt *pkt, u32_t data)
+ *
+ * @param pkt Network packet
+ *
+ * @return Amount of data to be read from current pkt cursor
```
Not "to be read", but "which can be read".

pfalcon commented:
@tbursztyka 

> Re-read my commit message: socket will need to be reworked a bit to actually get rid completely of such function.

Please also read to into my concern: you removed 300 lines of carefully crafted tests, which include packet fixture data, etc. We should write more tests, writing a test beyond 30 lines requires a bunch of effort, and instead you remove whole 300 lines of them! I'm not against particular functions to be downplayed and eventually removed, but please whenever possible find a way to preserve tests, even if the functions move to the tests themselves (like you do with a bunch of other tests in this PR, I don't know why those poor 300 lines got removes so cold-heartedly).

pfalcon commented:
Beyond concerns with the test removal, looks like a pretty good cleanup to me.

Commit:
net/icmpv4: Do not reuse echo request ttl for the echo reply

net_pkt_clone_new() sets the same attributes of the original packet to
the cloned one.

Fixes #13147

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/if: Use new net_pkt API for cloning in case of promiscuous mode

That was the last place where legacy net_pkt_clone() was used.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/pkt: Remove useless legacy net_pkt_clone() function

Now that the stack uses the new API from net_pkt for cloning, no need to
keep the legacy one around.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/pkt: Rename net_pkt_clone_new to net_pkt_clone

Now that legacy net_pkt_clone function has been removed, the new
function can be renamed accordingly.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/pkt: Remove useless legacy net_pkt_copy/_all functions

Now that the stack uses the new API from net_pkt for copying, no need
to keep the legacy one around.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/pkt: Rename net_pkt_copy_new to net_pkt_copy

Now that legacy net_pkt_clone function has been removed, the new
function can be renamed accordingly.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

jukkar commented on subsys/net/ip/net_private.h:
```diff
@@ -153,30 +153,3 @@ static inline void net_pkt_hexdump(struct net_pkt *pkt, const char *str)
 		buf = buf->frags;
```
Regarding the commit "net/private: Remove net_print_frags() function", I would like to keep the net_print_frags() function (we could rename it of course if needed). It is indeed not used anywhere but I have found it useful when debugging things.

pfalcon replied:
> I would like to keep the net_print_frags() function (we could rename it of course if needed).

I also wanted to express that concern, but not be accused of complaining too much, decided to concentrate on removed 300 lines of tests ;-)

tbursztyka replied:
@jukkar ok, though I guess it needs to use the log subsystem and not printk as well.

pfalcon replied:
> ok, though I guess it needs to use the log subsystem and not printk as well.

Why? Logging will keep choking unless set to "immediate" mode, and printk() rules anyway.

jukkar replied:
I originally used printk() as the function is suppose to print the output in many pieces that should be in one line. So using logger is no go here.

tbursztyka replied:
ok

jukkar commented on include/net/net_pkt.h:
```diff
@@ -2399,6 +2399,15 @@ static inline int net_pkt_write_le32_new(struct net_pkt *pkt, u32_t data)
 	return net_pkt_write_new(pkt, &data_le32, sizeof(u32_t));
```
In commit called "net/pkt: Add a function to know the amount of data to read left", the subject is not understandable. Could we say here "net/pkt: Add a function to know the amount of remaining data" instead

tbursztyka replied:
Sure. btw any idea on a better name for that function?

jukkar commented on include/net/net_pkt.h:
```diff
@@ -2399,6 +2399,15 @@ static inline int net_pkt_write_le32_new(struct net_pkt *pkt, u32_t data)
+ *
+ * @return Amount of data which can be read from current pkt cursor
+ */
+size_t net_pkt_data_to_read(struct net_pkt *pkt);
```
What about calling this func "net_pkt_data_remaining()" ?

tbursztyka replied:
yes, much better!

Commit:
net/pkt: Remove useless legacy net_pkt_pull function

Now that the stack uses the new API from net_pkt for pulling, no need
to keep the legacy one around.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/pkt: Rename net_pkt_pull_new to net_pkt_pull

Now that legacy net_pkt_pull function has been removed, the new
function can be renamed accordingly.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

jukkar commented on include/net/net_pkt.h:
```diff
@@ -2287,14 +2287,14 @@ int net_pkt_copy_new(struct net_pkt *pkt_dst,
 
 /**
- * @brief Clone pkt and its fragment chain.
+ * @brief Clone pkt and its buffer.
```
Should we say here "Clone pkt and its chain of buffers." instead?

Commit:
net/pkt: Remove useless legacy net_pkt_split function

Now that the stack does not use this function, it can be removed
safely.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/pkt: Remove useless legacy net_frag_linearize function

Now that the stack does not use this function, it can be removed
safely.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
drivers/ethernet: Switch SMSC911x to new net_pkt API

It was still using old allocator and net_pkt function to read/write the
data.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

tbursztyka commented:
I'll stop growing that PR.

There is more to come, but oriented on subsys/net/lib

Commit:
net/pkt: Remove useless legacy net_pkt_append_memset function

Now that the stack does not use this function, it can be removed
safely.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

tbursztyka commented:
Sorry, missed the net_pkt_append_memset() removal in order to remove net_frag_linearize() so I had to add a commit.

tbursztyka commented:
@pfalcon btw I don't have a board with SMSC911x  eth controller, could you check if my commit did not break the driver? As you are the original author of the driver.

pfalcon commented:
> @pfalcon btw I don't have a board with SMSC911x eth controller, could you check if my commit did not break the driver? As you are the original author of the driver.

I don't have either, it's for QEMU's BOARD=mps2_an385. There was a ready to use overlay in samples/net/echo_server/, but @jukkar removed that sample. Arggghhh! Will test.

tbursztyka commented:
@pfalcon yep now there is only sockets/echo apps. Socket is what users should use anyway. So add the overlay there.

jukkar reviewed (approved):
Lot of commits in one PR so I might have missed something. Anyway, this needs to be merged so +1

tbursztyka commented:
@nashif this PR would need to be merged

pfalcon commented on drivers/ethernet/eth_smsc911x.c:
```diff
@@ -531,71 +532,53 @@ static inline void smsc_wait_discard_pkt(void)
-		*buf32++ = SMSC9220->RX_DATA_PORT;
+		buf32 = SMSC9220->RX_DATA_PORT;
+
+		if (net_pkt_write_new(pkt, &buf32, sizeof(u32_t))) {
```
@tbursztyka, I'm sorry, but this is terribly slow to call a function for each 4 bytes of a packet. We need to get direct access to net buffer memory back.

The eth_smsc911x.c driver is all broken after your changes. There used to be CI tests for it, but those were removed by @jukkar without any extra thought. Just like you in this PR removed 350 lines of other tests. @nashif, here're good outcomes of the discussion at the network forum last week, where we talked about the need to write tests - I week after, a great number of tests is REMOVED.

jukkar replied:
Various sample applications were removed because of net_app removal. Unfortunately this removed some of the compilation tests at the same time. We should probably have had the same compilation tests in socket version of echo-server/client.

tbursztyka replied:
@pfalcon it's not like I did not ask you to look at it 6 days ago.

That said, we really need the overlays/config for legacy echo_server/client apps back yes. (we have the same issue for netusb drivers for instance).

pfalcon replied:
> @pfalcon it's not like I did not ask you to look at it 6 days ago.

And well, that's what I did, except that I have only 1 head and 2 hands, and I already find that I spend 50% of time in reviews normally, and 75% of time during release rush like we have, but I also need to do some things myself. And surely, I suggested to not haste with net_app removal, and outright warned against removal of 300 lines of tests in all these "release rush" PRs.

> That said, we really need the overlays/config for legacy echo_server/client apps back yes. (we have the same issue for netusb drivers for instance).

There're now https://github.com/zephyrproject-rtos/zephyr/issues/13481 and https://github.com/zephyrproject-rtos/zephyr/issues/13484 for both of these issues.
